### ISSUE TITLE
[Bug]: error in kube_ingress when trying to enable ingress for kube_monitoring

### ISSUE THREAD

**Issue by mschnee** [Panfactum/stack#173]
### Prior Search

- [X] I have already searched this [project's issues](https://github.com/Panfactum/stack/issues) to determine if a bug report has already been made.

### What happened?

On `edge.24-10-21`, we have newly deployed `kube_monitoring` successfully with ingress disabled.  When we enable ingress and attempt to `init`, `plan`, or `apply`, we receive the attached following error.   Our production environment already had monitoring deployed from a previous release.  

We are considering pinning `kube_monitoring` to an earlier edge version to attempt to deploy it (though the module is in alpha, having Grafana and Loki is still incredibly useful), but are holding off in order to provide more details if asked.

### Steps to Reproduce

Deploy kube_monitoring with ingress disabled.

Attempt to init, plan, or apply kube_monitoring with ingress enabled.

### Relevant log output

```shell
╷
│ Error: Invalid for_each argument
│
│   on ../kube_ingress/main.tf line 275, in resource "kubectl_manifest" "ingress":
│  275:   for_each = var.cdn_mode_enabled ? local.ingress_configs_with_cdn : local.ingress_configs
│     ├────────────────
│     │ local.ingress_configs will be known only after apply
│     │ local.ingress_configs_with_cdn will be known only after apply
│     │ var.cdn_mode_enabled is false
│
│ The "for_each" map includes keys derived from resource attributes that
│ cannot be determined until apply, and so OpenTofu cannot determine the full
│ set of keys that will identify the instances of this resource.
│
│ When working with unknown values in for_each, it's better to define the map
│ keys statically in your configuration and place apply-time results only in
│ the map values.
│
│ Alternatively, you could use the -target planning option to first apply
│ only the resources that the for_each value depends on, and then apply a
│ second time to fully converge.
```

**Comment by mschnee** [comment]
I have not attempted to deploy a net new module that uses ingress yet, but will report back if I am able to.

**Comment by wesbragagt** [comment]
I've attempted to pin to an older version and kept getting 403. Still getting the same issue.

**Comment by fullykubed** [comment]
@mschnee @wesbragagt 

> I have not attempted to deploy a net new module that uses ingress yet, but will report back if I am able to.

Is it possible to do this and report back?

**Comment by wesbragagt** [comment]
I'm unsure what steps to follow regarding a net new module that uses ingress. Could you clarify? I'm open to try it.

**Comment by wesbragagt** [comment]
@fullykubed I cannot deploy that module with ingress_enabled set to true. I get the same error reported.

**Comment by wesbragagt** [comment]
I've destroyed the module and applied it again, landed on the same issue.

**Comment by mschnee** [comment]
I was sort-of able to get this working:
```
terragrunt apply -target 'module.ingress[0].kubectl_manifest.ingress_cert'
```

Then

```
terragrunt apply -target 'module.ingress[0].kubectl_manifest.ingress'
```

And then it errors on authenticating proxy, so I'm doing the same:

```
terragrunt apply -target 'module.authenticating_proxy[0].module.ingress.kubectl_manifest.ingress_cert'`
```

Then 
```
terragrunt apply -target 'module.authenticating_proxy[0].module.ingress.kubectl_manifest.ingress_'
```

**Comment by mschnee** [comment]
I sus

**Comment by mschnee** [comment]
I will make a new module first-party module today (a test webapp) to further test and report back.

**Comment by fullykubed** [comment]
Ok. I will also look into it this afternoon and see if it can be resolved before today's release.

**Comment by fullykubed** [comment]
@mschnee @wesbragagt 

Are you by any chance mixing versions between the local dev shell and the `kube_monitoring` module? Or are you adding any other non-standard customizations to the stack?

I am not able to replicate this issue by running `terragrunt destory --target module.ingress` on the `kube_monitoring` deployment in our test cluster and then re-applying. Additionally, I am not eager to delete the module from the reference cluster b/c we are using the stored data for testing.

I have tried creating a net new module using the latest version of `kube_ingress` and was also not able to reproduce this error. I have tried with both `cdn_mode_enabled` set to `false` and `true`.
